modules/account-lib/src/coin/baseCoin/baseTransactionBuilder.ts

```diff
@@ -140,6 +140,51 @@ export abstract class BaseTransactionBuilder {
     return this._coinConfig.name;
   }
 
+  /**
+   * Verified validity windows params if them exist and return a valid validity windows.
+   * Unit param must be specified
+   * If params are not consistent, default params will be return based on firstValid and minDuration
+   * @param {ValidityWindow} params validity windows parameters to validate.
+   * @param {String} params.unit Parameter that could be 'blockheght' or 'timestamp'
+   * @param {Number} [params.minDuration] Optional - Minimum duration of the window
+   * @param {Number} [params.maxDuration] Optional - Maximum duration of the window
+   * @param {Number} [params.firstValid] Optional - First valid value
+   * @param {Number} [params.lastValid] Optional - Last valid value
+   * @returns {ValidityWindow} verified validity windows or default values
+   */
+  getValidityWindow(params: ValidityWindow): ValidityWindow {
```

Comment:
not 100% sure how this is going to be used.

one thing that sticks out is that this function is in the builder, but doesn't actually mutate the underlying transaction in some way.

as an example, EOS supports setting a transaction expiration time (e.g. `lastValid = now + '1 hour'`). how can i use this function to set that expiration time?

Reply:
It is not clear to me either, but base on comments, this method:
- should return some default values if params are empty
- should verify params if some or all of them are present
- should build new default params based on the existent ones if they are not consisten.

In the example you mentioned, **_now_** could be the `firstValid` and **_1 hour_** could be the `maxDuration`.
Anyway, `defaultDuration` could be differenciate for `blockheight` and `timestamp` and it could be differenciated an `initialValue` to be **_now_** for `timestamp` an **_0_** for `blockheight`